### PR TITLE
Added Phase4 to Assignment 3

### PR DESCRIPTION
**Pull request by team6-pims**

### CODE REVIEW

File: CST338/Seperate files/assig3.java [hunk 1]
```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
...
+            validInput = true;
+
+      } while (!validInput);
       
+      // Instantiate game objects
+      Deck deck = new Deck();                   // Single deck, not shuffled
+      Hand players[] = new Hand[numPlayers];    // Array containing player hands
+      
+      // Give each player a hand
+      for (int i = 0; i < numPlayers; i++)
```
Comment (ialej001) [inline CST338/Seperate files/assig3.java]: use for each loops when possible

for (Hand player: players) {
}

File: CST338/Seperate files/assig3.java [hunk 2]
```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
...
+         System.out.println("Player " + i + " Hand: ");
+         System.out.println(players[i-1].toString());
+      }
+
+      // Get cards back from players
+      for (int i = 0; i < players.length; i++)
+         players[i].resetHand();
+      
+      // Reset and shuffle deck
+      deck.init(1);
```
Comment (ialej001) [inline CST338/Seperate files/assig3.java]: shouldn't this be 
deck.init(numPlayers)
or even better, 
deck = new Deck(numPlayers)

Comment (team6-pims) [reply CST338/Seperate files/assig3.java]: The function prototype for init is init(numPacks).

File: CST338/Seperate files/assig3.java (hunk 1 above)
Comment (conocolo) [reply CST338/Seperate files/assig3.java]: How would I use for each loops to assign each item in the array a new Hand()?

**Comment by conocolo** [comment]
> good stuff, condense comments, make suggested changes, and we're golden

What comments do you suggest condensing?

File: CST338/Seperate files/assig3.java (hunk 1 above)
Comment (ialej001) [reply CST338/Seperate files/assig3.java]: for (Hand player: players) {
   player = new Hand()
}

the style is
for (Object-type elementName: array) {
   do stuff to elementName
}

File: CST338/Seperate files/assig3.java [hunk 3]
```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
...
+      do
+      {
+         // Request user input
+         System.out.print("Enter number of players in game (1-10): ");
+         numPlayers = userInput.nextInt();
+
+         // Verify input
+         if (numPlayers >= MIN_PLAYERS && numPlayers <= MAX_PLAYERS)
+            
+            // Break loop if user input is good
```
Comment (ialej001) [inline CST338/Seperate files/assig3.java]: merge with line 133, delete blank lines.

File: CST338/Seperate files/assig3.java [hunk 4]
```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
...
+      Hand players[] = new Hand[numPlayers];    // Array containing player hands
+      
+      // Give each player a hand
+      for (int i = 0; i < numPlayers; i++)
+         players[i] = new Hand();
+      
+      // Get number of cards in deck and store it
+      numCards = deck.topCard();                
+
+      // Deal cards to players
```
Comment (ialej001) [inline CST338/Seperate files/assig3.java]: all comments in loop should be here, block comment if necessary

File: CST338/Seperate files/assig3.java [hunk 5]
```diff
@@ -108,7 +113,94 @@ public static void main(String[] args) {
...
+      // Get cards back from players
+      for (int i = 0; i < players.length; i++)
+         players[i].resetHand();
+      
+      // Reset and shuffle deck
+      deck.init(1);
+      deck.shuffle();
+      System.out.println("\n------Shuffled Deck------");
+
+      // Deal cards to players
```
Comment (ialej001) [inline CST338/Seperate files/assig3.java]: merge comments in loop, block if necessary

File: CST338/Seperate files/assig3.java (hunk 3 above)
Comment (conocolo) [reply CST338/Seperate files/assig3.java]: Sure thing

File: CST338/Seperate files/assig3.java (hunk 4 above)
Comment (conocolo) [reply CST338/Seperate files/assig3.java]: Sure thing

File: CST338/Seperate files/assig3.java (hunk 5 above)
Comment (conocolo) [reply CST338/Seperate files/assig3.java]: Sure thing

File: CST338/Seperate files/assig3.java (hunk 1 above)
Comment (conocolo) [reply CST338/Seperate files/assig3.java]: The code snippet posted doesn't seem to work. The players array is not being assigned a new hand as intended. Am I missing something?

**Pull request merged**